lib/route/link/macvlan.c

```diff
@@ -332,6 +478,166 @@ uint16_t rtnl_link_macvlan_get_flags(struct rtnl_link *link)
 	return mvi->mvi_flags;
 }
 
+/**
+ * Get number of MAC-Addr for MACVLAN device in source mode
+ * @arg link		Link object
+ *
+ * @return number of addresses on success or a negative error code.
+ */
+int rtnl_link_macvlan_count_macaddr(struct rtnl_link *link)
+{
+	struct macvlan_info *mvi = link->l_info;
+
+	IS_MACVLAN_LINK_ASSERT(link);
+
+	if (!(mvi->mvi_mask & MACVLAN_HAS_MODE) ||
+	    (mvi->mvi_mode != MACVLAN_MODE_SOURCE)) {
+		APPBUG("Only macvlan source mode links can have a list of "
+		       "source mac addresses. Set mode \"source\" first.");
+		return -NLE_OPNOTSUPP;
+	}
+
+	if (!(mvi->mvi_mask & MACVLAN_HAS_MACCOUNT)) {
+		APPBUG("You need to read the list of macs from the macvlan "
+		       "device before.");
+		return -NLE_OPNOTSUPP;
+	}
+
+	return mvi->mvi_maccount;
+}
+
+/**
+ * Get configured remote MAC-Addr from MACVLAN device in source mode
+ * @arg link		Link object
+ *
+ * @return cloned nl_addr struct (to be freed) or NULL on error
+ */
+struct nl_addr *rtnl_link_macvlan_get_macaddr(struct rtnl_link *link, int idx)
+{
+	struct macvlan_info *mvi = link->l_info;
+
+	IS_MACVLAN_LINK_ASSERT(link);
+
+	if (!(mvi->mvi_mask & MACVLAN_HAS_MODE) ||
+	    (mvi->mvi_mode != MACVLAN_MODE_SOURCE)) {
+		APPBUG("Only macvlan source mode links can have a list of "
+		       "source mac addresses. Set mode \"source\" first.");
+		return NULL;
```

Comment:
maybe rtnl_link_macvlan_get_macaddr() also should return an "int" error-code.

And maybe just get away with these assertions. If the link has the wrong mode, then it should return just return -NLE_INVAL.